### ISSUE TITLE
SignalException cop incorrectly flags "Use `fail` instead of `raise` to signal exceptions

### ISSUE THREAD

**Issue by josh-h** [rubocop/rubocop#909]
With the following code snippet:

``` ruby
def test
  5/0
rescue ZeroDivisionError
  # handle error
rescue
  raise
end
```

The SignalException cop flags the error:

> Use `fail` instead of `raise` to signal exceptions.

Since we are re-raising an exception this should be valid. Simplifying the code further by removing the first rescue stops the cop from flagging an issue. So, this only becomes a problem when two rescues are used, and one raise's the exception.

I created the following spec, but am too unfamiliar with the codebase to provide a fix at this time:

``` ruby
    it 'accepts raise in def with multiple rescues' do
      inspect_source(cop,
                     ['def test',
                      '  fail',
                      'rescue StandardError',
                      '  # handle error',
                      'rescue Exception',
                      '  raise',
                      'end'])
      expect(cop.offenses).to be_empty
    end
```
